feat: Added the new filter to get users by country

GnesisVSS opened the pull request:
Added a new filter to retrieve user data (excluding passwords) based on country, restricted to admin roles only.

GnesisVSS committed:
feat: Added the new filter to get users by country

GnesisVSS committed:
feat: Added the new filter to get users by country

kamivillabla committed:
feat: added validations for first name, last name and e-mail and date of birth

kamivillabla committed:
refactor: simplify user creation response

kamivillabla committed:
refactor: move validation logic from controller to service

kamivillabla committed:
feat: Made birthdate field mandatory in user model and added validation comments in user service

LuchoSerruya commented on controllers/adminController.js:
```diff
@@ -11,5 +11,17 @@ export const adminController = {
+      const { country } = req.body;
+      try {
+          const users = await adminService.filterByCountry(country);
+          if (!users) res.status(401).send({ massage: "No users where found" })
```
El código de estado HTTP 401 es para indicar que una petición no puede ser realizada porque el usuario no está autenticado. En este caso, que no haya usuarios no significa que el usuario no esté autenticado, es más, ni siquiera puede representar un error. Devolver un arreglo sin datos puede no ser un error (lo dejo a su criterio)

GnesisVSS replied:
corregido!

LuchoSerruya commented on services/adminService.js:
```diff
@@ -35,5 +35,26 @@ export const adminService = {
+  filterByCountry: async (country) => {
+    try {
+        const usersFoundByCountry = await User.findAll({where: {country: country}})
+        .then(users => users.map(u => ({
```
Si usan `await`, no hace falta usar el `.then`. Pueden hacer lo siguiente

```javascript
const usersFoundByCountry = await User.findAll({...});
const usersData = usersFoundByCountry.map(...);
```

GnesisVSS replied:
corregido!

kamivillabla committed:
Merge pull request #7 from alkemyTech/feat/user-registration-validations

Feat/user registration validations

GnesisVSS committed:
Merge branch 'feat/get-users-by-country' of https://github.com/alkemyTech/CENCO-MIT-Back-T1 into feat/get-users-by-country

GnesisVSS committed:
fix: fixed the return of the data and await function

Fix issue in searchUsersByCountry function in adminController to return search results reliably, handling cases with no matching users, and deleted .then in adminService.
